ffmpeg_utils.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+from datetime import datetime, timedelta
+import subprocess
+
+from constants import RAW_VIDEOS_FRAMERATE
+
+
+def subprocess_run_with_guardfile(cmd: list[str], guardfile_path: Path):
+    """
+    Runs a subprocess command, and writes the command to a guardfile, which is deleted after subprocess runs
+    successfully as a way to monitor whether there was a crash mid-processing, or a forced stop.
+
+    """
+    running_message: str = "Running: " + " ".join(cmd)
+    with open(guardfile_path, "w") as f:
+        f.write(running_message)
+    subprocess.run(cmd, check=True)
+    guardfile_path.unlink()
+
+
+def get_video_duration_in_seconds(file_path: Path) -> float:
+    cmd = [
+        "ffprobe",
+        "-v",
+        "error",
+        "-show_entries",
+        "format=duration",
+        "-of",
+        "default=noprint_wrappers=1:nokey=1",
+        file_path,
+    ]
```

Comment:
I usually like to format the command to match flags with values
``` python
# fmt: off
    cmd = [
        "ffprobe",
        "-v", "error",
        "-show_entries", "format=duration",
        "-of", "default=noprint_wrappers=1:nokey=1",
        file_path,
    ]
# fmt: on
```
and for the one below
```python
    cmd = [
        "ffprobe",
        "-hide_banner",
        "-show_streams",
        "-i", str(video_path),
    ]
```

Reply:
Fixed in 124b5be58b35d448a82a245bb968c691206c83f8